### PR TITLE
[search] Filter villages by distance from viewport center/user position.

### PR DESCRIPTION
**Pull request by tatiana-yan**

### CODE REVIEW

**Comment by tatiana-yan** [comment]
JTALL

**Comment by tatiana-yan** [comment]
JTALL

File: search/geocoder.cpp [hunk 1]
```diff
@@ -766,8 +780,24 @@ void Geocoder::FillVillageLocalities(BaseContext const & ctx)
...
+    if (!m_context->GetCenter(l.m_featureId, center))
+      center = feature::GetCenter(*ft);
+
+    vector<m2::PointD> pivotPoints = {m_params.m_pivot.Center()};
+    if (m_params.m_position)
+      pivotPoints.push_back(*m_params.m_position);
+    auto const mwmType = m_context->GetType();
+    CHECK(mwmType, ());
+    if (!mwmType->m_containsMatchedState &&
+        !any_of(pivotPoints.begin(), pivotPoints.end(), [&](auto const & p) {
```
Comment (mpimenov) [inline search/geocoder.cpp]: all_of?

File: search/search_integration_tests/processor_test.cpp [hunk 2]
```diff
@@ -2635,5 +2635,111 @@ UNIT_CLASS_TEST(ProcessorTest, FilterStreetPredictions)
     TEST(ResultsMatch(request.Results(), rules), ());
   }
 }
+
+UNIT_CLASS_TEST(ProcessorTest, FilterVillages)
+{
+  TestState moscowRegion(m2::PointD(10.0, 10.0), "Moscow Region", "en");
+  // |moscowRegion| feature shold belong to MoscowRegion with some margin due to mwmPointAccuracy.
```
Comment (mpimenov) [inline search/search_integration_tests/processor_test.cpp]: s/shold/should/

**Comment by tatiana-yan** [comment]
JTLINUX

File: search/geocoder.cpp (hunk 1 above)
Comment (tatiana-yan) [reply search/geocoder.cpp]: done

File: search/search_integration_tests/processor_test.cpp (hunk 2 above)
Comment (tatiana-yan) [reply search/search_integration_tests/processor_test.cpp]: done

**Pull request merged**